bee-ternary/src/num_conversions.rs

```diff
@@ -83,17 +82,22 @@ impl From<i64> for TritBuf<T1B1Buf> {
     }
 }
 
+pub enum TritsI64ConversionError{
+    AbsValueTooBig
+}
+
 //TODO - generalize over all encodings
 //Impl copied from:
 // [https://github.com/iotaledger/iota_common/blob/1b56a5282933fb674181001630e7b2e2c33b5eea/common/trinary/trit_long.c#L31]
-impl From<TritBuf<T1B1Buf>> for i64 {
-    fn from(trits: TritBuf<T1B1Buf>) -> i64 {
+impl TryFrom<TritBuf<T1B1Buf>> for i64 {
+    type Error = TritsI64ConversionError;
+    fn try_from(trits: TritBuf<T1B1Buf>) -> Result<Self,Self::Error> {
         if trits.len() == 0 {
-            return 0;
+            return Ok(0);
```

Comment:
Shouldn't it actually be an error?

Reply:
Debatable..
but let's return an error nonetheless